http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnector.java

```diff
@@ -0,0 +1,77 @@
+package org.triplea.http.client.web.socket;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.java_websocket.client.WebSocketClient;
+
+/**
+ * Class to handle the task of creating a websocket connection asynchronously and then provides a a
+ * blocking method to allow us to wait for the connection to be established. Connecting to a local
+ * server, a websocket connection can take several hundred milliseconds.
+ */
+class WebSocketConnector {
+  private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = 5000;
+
+  private final WebSocketClient webSocketClient;
+  private final int connectTimeoutMillis;
+
+  /**
+   * We use this to create a single blocking task that is our connection attempt. Eventually to
+   * block for completion, we'll await completion of tasks on this queue.
+   */
+  private ExecutorService executorService;
+
+  /**
+   * This is the thread that is attempting to connect to server. Returns true if connection is
+   * successful otherwise false.
+   */
+  private Future<Boolean> connectionThread;
+
+  WebSocketConnector(final WebSocketClient webSocketClient) {
+    this(webSocketClient, DEFAULT_CONNECT_TIMEOUT_MILLIS);
+  }
+
+  @VisibleForTesting
+  WebSocketConnector(final WebSocketClient webSocketClient, final int connectTimeoutMillis) {
+    this.webSocketClient = webSocketClient;
+    this.connectTimeoutMillis = connectTimeoutMillis;
+  }
+
+  void initiateConnection() {
+    executorService = Executors.newSingleThreadExecutor();
+    connectionThread = executorService.submit((Callable<Boolean>) webSocketClient::connectBlocking);
+    executorService.shutdownNow();
+  }
+
+  void waitUntilConnectionIsOpen() {
+    try {
+      executorService.awaitTermination(connectTimeoutMillis, TimeUnit.MILLISECONDS);
+      if (!connectionThread.get() || !webSocketClient.isOpen()) {
+        throw new CouldNotConnect();
+      }
+    } catch (final InterruptedException | ExecutionException e) {
+      throw new CouldNotConnect(e);
```

Comment:
Indeed. Also throwing is probably not the right behavior. We likely want to no-op, an exception would show a prompt to a user rather than letting the application abort or proceed to terminate. Updated in: b845467cd, sets interrupt and does a no-op if current thread is interrupted.